### PR TITLE
Add Paperless and Recoll document search integrations

### PR DESCRIPTION
**Pull request by aleksclark**
## Summary
- Add Paperless-ngx document search, metadata, download, update, deletion, and safe test-upload tools.
- Add Recoll WebUI search backed by its live HTML query endpoint.
- Document isolated development smoke testing and integration configuration.

## Verification
- `make ci`
- Isolated live smoke tests against Paperless-ngx and Recoll WebUI

💘 Generated with Crush

### CODE REVIEW

File: integrations/paperless/paperless.go [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+	req.Header.Set("Authorization", "Token "+p.token)
+	if body != nil {
+		req.Header.Set("Content-Type", "application/json")
+	}
+	resp, err := p.client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("make Paperless request: %w", err)
+	}
+	defer func() { _ = resp.Body.Close() }()
+	data, err := io.ReadAll(resp.Body)
```
Comment (acmacalister) [inline integrations/paperless/paperless.go:110]: These response bodies are read with unbounded `io.ReadAll`. That is especially painful on `paperless_download_document`, where a multi‑MB PDF can land entirely in memory, and it is out of step with newer adapters that cap reads (`io.LimitReader` + a max response size).

Worth applying the same limit in both `doRequest` and `doMultipartRequest` (and preferably a tighter cap on the download path).

File: integrations/paperless/paperless.go [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+func downloadDocument(ctx context.Context, p *paperless, args map[string]any) (*mcp.ToolResult, error) {
+	id, err := documentID(args)
+	if err != nil {
+		return mcp.ErrResult(err)
+	}
+	data, err := p.get(ctx, fmt.Sprintf("/api/documents/%d/download/", id))
+	if err != nil {
+		return mcp.ErrResult(err)
+	}
+	return &mcp.ToolResult{Data: string(data)}, nil
```
Comment (acmacalister) [inline integrations/paperless/paperless.go:287]: `string(data)` will mangle binary originals (PDFs, images) and dump raw bytes straight into the tool result. Drive solved this with a small envelope: size cap, `content_type`, inline text for `text/*`, and `content_base64` otherwise (`gdrive` `downloadEnvelope`).

Something in that shape would make downloads actually usable for agents instead of producing corrupted UTF‑8 noise.

File: integrations/paperless/paperless.go [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+
+type paperless struct {
+	token   string
+	baseURL string
+	client  *http.Client
+}
+
+// New creates a Paperless-ngx integration.
+func New() mcp.Integration {
+	return &paperless{client: &http.Client{}}
```
Comment (acmacalister) [inline integrations/paperless/paperless.go:42]: `&http.Client{}` has no timeout, so a hung Paperless instance can block a tool call indefinitely. Recoll in this same PR sets `Timeout: 30 * time.Second` — matching that here would be consistent and safer.

File: integrations/recoll/tools.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+	},
+}
+
+var dispatch = map[mcp.ToolName]handlerFunc{
+	"recoll_search": search,
+}
+
+func search(ctx context.Context, r *recoll, args map[string]any) (*mcp.ToolResult, error) {
+	reader := mcp.NewArgs(args)
+	query := reader.Str("query")
```
Comment (acmacalister) [inline integrations/recoll/tools.go:34]: `Str("query")` accepts an empty string, so a missing/blank query still hits the WebUI. Paperless search rejects empty query explicitly; same guard here would avoid a pointless HTML round-trip and a confusing empty result set.

File: integrations/recoll/tools.go [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+			return
+		}
+		if hasClass(n, "result") {
+			response.Results = append(response.Results, searchResult{
+				Path:     textForClass(n, "path"),
+				MIMEType: decodeMIMETypes(textForClass(n, "meta")),
+			})
+			return
+		}
+		if n.Data == "p" && response.Count == 0 {
```
Comment (acmacalister) [inline integrations/recoll/tools.go:82]: If the count paragraph markup changes, `Count` stays 0 even when `Results` is non-empty. Falling back to `len(response.Results)` (or setting count from results when the regex misses) would keep the payload coherent across WebUI tweaks.

**Review by acmacalister — COMMENTED** [review]
Nice pair of integrations — wiring, dispatch parity, compaction specs, and config/env defaults all look solid, and CI is green across build/test/lint/security. A few follow-ups on Paperless downloads and a couple of Recoll edge cases below (PR is already merged, so these are post-merge notes).
